### DISCUSSION TITLE
Caddy su QNAP via Docker

### DISCUSSION THREAD

**Discussion by leadermar** [indomus/forum#1384] [GENERALE - Discussioni generiche]
Ciao Ragazzi, ho installato Home assistant core su qnap attraverso Docker, adesso stavo cercando di installare DUCK DNS con Caddy, ma non riesco a portare l'operazione a buon fine.  Come metodo ho utilizzato HTTP Challange, con porta TCP 80 ( la porta 443 invece non va), e sembrerebbe comunicare, ma non mi riporta nessun certificato. ecco i LOG 
![Immagine 2022-01-24 124447log](https://user-images.githubusercontent.com/98328267/150794502-d30bbb07-e6e1-482d-9cbc-39b64851b853.png)

Dove sbaglio?

Grazie a tutti

**Comment by snakuzzo** [comment]
Innanzi tutto ti correggo...e non per fare il precisino, ma perché bisogna utilizzare la terminologia corretta...non "installi" nessun Duckdns...semplicemente configuri Caddy per la gestione dei certificati e per esporre le tue applicazioni in https
Ad ogni modo nel log non vedo da nessuna parte il tuo FQDN (che nel caso sarebbe meglio oscurare), quindi molto probabilmente hai sbagliato a montare i volumi...nello specifico il flle `Caddyfile`

**Reply by leadermar** [reply]
Grazie per la rapida risposta.

Ti allego screen, a me sembra tutto ok, ho seguito la guida su indomus
![Immagine 2022-01-24 150022](https://user-images.githubusercontent.com/98328267/150796853-77ecadbe-6356-4924-b042-33ba8b079d48.png)
![Immagine 2022-01-24 150129](https://user-images.githubusercontent.com/98328267/150796857-71a4e141-cbd9-49c4-9cba-7960e9c1a206.png)

**Reply by snakuzzo** [reply]
il Caddyfile deve essere "SENZA" estensione ?

**Reply by leadermar** [reply]
ho tolto l'estensione, nulla è cambiato. Avendo io no IP configurato sul mio router, ho impostato un dynamic DNS. Non vorrei che il problema fosse li, oppure se posso sorvolare caddy, ed utilizzare il DDNS di NO IP.

Grazie ancora

**Reply by snakuzzo** [reply]
Non ha importanza quale servizio DDNS stai usando. Con le challenge http e https puoi usare quello che ti pare. Resta il fatto che se NON vedi nel log che gestisce il dominio che hai impotsato nel Caddyfile, vuol dire che hai sbagliato a mappare i volumi... poi tocca capire se hai sbagliato ad indicare le directory o hai sbagliato a creare directory e file.

La seconda immagine che hai postato  mi sembra strana. Sicuro di aver mappato correttamente le directory come indicato in guida ?

![acq](https://user-images.githubusercontent.com/7205371/150840588-c77ae481-392e-4759-9b59-569441ee5ac8.png)

**Reply by snakuzzo** [reply]
Posso solo dirti che la guida è corretta...testata e ritestata (non solo da noi).
Se ti leggi un attimino come funzionano i mount dei volumi su Docker, magari la causa riesci a trovarla da solo

**Comment by leadermar** [comment]
Ho risolto, effettivamente sbagliavo a mappare la directory di etc config. Sembrerebbe tutto ok, ma quando accedo da remoto con il FQDN, riesco ad entrare ma dopo un po' appare connessione persa, per poi riprendersi. Dall'app Android invece non riesce ad entrare.

Cosa ho sbagliato?

**Reply by snakuzzo** [reply]
noi non possiamo saperlo...ti sei ricordato di impostare `trusted proxies` nel blocco `http` del `configuration.yaml` di Home Assistant ?

**Reply by LZDany** [reply]
Ciao leadermar, scusa se approfitto della tua discussione. Ho lo stesso identico problema ma non riesco a risolverlo... come hai rimappato la directory?
Grazie in anticipo!

**Reply by snakuzzo** [reply]
Le directory vanno mappate come da guida, a meno che non abbiate scelto percorsi diversi.
E se il funzionamento è altalenante, non credo possa dipendere dai passi seguiti in guida per l'installazione.
Dovreste leggere i log, non basarvi sul semplice comportamento che si riscontra da app o da web

**Reply by LZDany** [reply]
ciao snakuzzo,
ho fatto tutto pari pari alla guida ma ho lo stesso problema di leadermar. Il contenitore è attivo ma non mi creano le chiavi e le cartelle... sul log del qnap "container station" non trovo niente.
Tieni presente che per me è un mondo nuovo quindi potrei aver fatto qualche errore anche banale... mi sono collegato a questa discussione perchè il testo della console del container incollato sopra da leadermar è identica alla mia

**Reply by snakuzzo** [reply]
Sì ma capisci che in base a quello che dici, l'unica cosa che uno può dirti è di ricontrollare come hai mappato le directory (e infatti anche l'altro utente poi si è reso conto che aveva sbagliato).
Quindi...ricontrolla BENE

**Reply by LZDany** [reply]
![caddy1](https://user-images.githubusercontent.com/100022334/154797316-d51c7ade-825e-41c5-9b68-c87366d72bea.PNG)

**Reply by LZDany** [reply]
![caddy2](https://user-images.githubusercontent.com/100022334/154797368-6ea80b97-7570-4e0d-a08d-04d0337f74bc.PNG)

**Reply by LZDany** [reply]
così è come ho le cartelle create

**Reply by snakuzzo** [reply]
E il mapping?

**Reply by LZDany** [reply]
il caddyfile è copiato incollato. Il contenitore è strutturato identico alla guida :(

**Reply by snakuzzo** [reply]
Fai vedere il mapping ed il log di Caddy

**Reply by LZDany** [reply]
perdona a questo punto la mia ignoranza. Il mapping? come lo vedo? A questo punto credo che nella mia testa il concetto di mapping sia sbagliato

**Reply by LZDany** [reply]
credo di aver risolto. Avevo una maiuscola di troppo. Ora ha creato le cartelle anche se non vedo gli stessi file che sono indicati nella guida... EDIT bastava cercare meglio....
Ho fatto un passo avanti. Grazie @snakuzzo !
